### ISSUE TITLE
Ability to upload and view code coverage

### ISSUE THREAD

**Issue by spartanglady** [reportportal/reportportal#697]
**Describe the solution you'd like**
... Is there any chance or plans to have the ability to add code coverage to report portal and track the history of it. ?

**Is your feature request related to a problem? Please describe.**
...

**Comment by evjlobanova** [comment]
@spartanglady  We are thinking about test plan coverage and requirements coverage.
And could you please describe your idea?
What do you mean under history? 
What is the main business need? 
Who is the main users for this feature?

**Comment by sspieker-cc** [comment]
This business requirement (for my company) is as follows:
Code coverage for our projects come in various flavors - .Net, .Net-core, Java, etc.
We prefer to utilize one tool to evaluate the increase in tests as well as the code coverage for the overall per project.  Although there are a number of environments (Jenkins, developer workstations, etc.) that could execute these tests and coverage tools, we are looking for a unified way to present the resulting data to interested parties.

Today the build pipeline produces a number of artifacts that are consumed in different ways with different tools and the metrics are not presented in a concise way.  Each of the consuming tools present methods to export their parsed information then these various tools have to have their data aggregated into yet another tool (often times Google Sheets or Excel) in order to massage the results into something that is more cohesive.

The main users of this test and coverage data are:

Developers (measuring the projects that they are involved with) for sake of knowing that new features have adequate coverage and tests and for metrics regarding monetary incentives.

QA Engineers - ensuring that all tests are being executed as well as monitoring of issues generated, coded and resolved.  They also plan new test scenarios in order to cover the areas of code that are being missed/overlooked in order to uncover infrequently used scenarios/conditions.

Product Managers - Following goals being set by upper management as well as team goals for continuously increasing test scenarios and coverage as well

Senior Management - Following progress of business initiatives and project and budget planning.  These metrics are used to identify velocity of teams and how to slice budgets for upcoming initiatives.

Code coverage coupled with test results is integral for today's modern cloud based businesses and an ever-increasing code base, test scenarios and test code coverage impact our development team at every level.

**Comment by DzmitryHumianiuk** [comment]
we have a separate open sourced tool for this: https://drill4j.github.io/

ReportPortal will retrieve coverage/mapping information from it going forward.
but it's a far far away backlog.

**Comment by gdroes** [comment]
Hi, is there any progress or at least a timeline available for the integration of Code Coverage to ReportPortal

**Comment by DzmitryHumianiuk** [comment]
@gdroes not really. still a ton of functionality to add for the general testing purposes.

**Comment by gdroes** [comment]
we are currently planning to switch to reportportal. Unfortunately, we would then lose our code coverage solution. We are therefore very interested in a solution

**Comment by DzmitryHumianiuk** [comment]
@gdroes what do you use for the coverage?

**Comment by gdroes** [comment]
@DzmitryHumianiuk we are using JaCoCo Coverage

**Comment by sspieker-cc** [comment]
The report portal is a wonderful tool to allow QA and Development staff to have a single source of truth for test results as well as for coverage information.  We have a series of integration and E2E test "pacakges" that are containerized and run within a Kubernetes environment.  When these tests are running, we do not have visibility to the output of the tests and therefore must rely upon the reporting aspects of the ReportPortal to summarize these results.

Jenkins provides a plugin that can be used to summarize the jacoco coverage reporting information which unfortunately is the only tool that we have at the moment.  Using these two systems does not make for an ideal and cohesive connection between a test run and the coverage information since we do not have an easy path of connecting these two results together.

It has been a part of our plan to perform incremental testing so that we can get a variety of metrics summarized: Unit Test coverage, Acceptance Test coverage, Integration, regression and e2e values.  This information will allow the QA team to better understand and know where to apply more time with additional testing to help increase coverage and better understand where this highest impact is applied and where there may be a good deal of overlap (when the coverage numbers do not increase as much as they were expected to do so).

Although the above use-case is ideal for us, it would be wonderful to have an aggregate result of all tests being presented in the same tool that the tests are reported within and valid linkage between the run and the results.

**Comment by sspieker-cc** [comment]
Perhaps the Jenkins Jacoco tooling could be adapted for the ReportPortal tool?

**Comment by DzmitryHumianiuk** [comment]
@sspieker-cc have you tried [Drill4J](https://drill4j.github.io/)? 
It's another tool in my portfolio, designed specifically for what you're referring to. 
Give it a try! 
It's based on Jacoco reports, it utilizes data to provide actionable feedback for both QA and development teams.

<img width="1486" alt="image" src="https://github.com/reportportal/reportportal/assets/11332788/432e93f3-1950-4280-8aea-e9c5e5ebd47f">
